java/src/main/java/com/databricks/mlflow/LoaderModule.java

```diff
@@ -0,0 +1,92 @@
+package com.databricks.mlflow;
+
+import com.databricks.mlflow.models.Model;
+import com.databricks.mlflow.sagemaker.Predictor;
+import com.databricks.mlflow.sagemaker.PredictorLoadingException;
+import com.databricks.mlflow.Flavor;
+
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * A generic loader for encapsulating flavor-specific model deserialization logic.
+ * By extending {@link com.databricks.mlflow.LoaderModule}, models of a specific flavor
+ * can be loaded as generic {@link com.databricks.mlflow.sagemaker.Predictor} objects.
+ * This allows tools, such as model containers, to use the models for inference
+ */
+public abstract class LoaderModule<T extends Flavor> {
+    /**
+     * Loads an MLFlow model as a generic predictor that can be used for
+     * inference
+     *
+     * @throws {@link com.databricks.mlflow.sagemaker.PredictorLoadingException} if
+     * failures are encountered while attempting to load the model from the specified configuration.
+     * This is a generic exception for all loader failure cases
+     */
+    public Predictor load(Model model) {
+        Optional<T> flavor = model.getFlavor(getFlavorName(), getFlavorClass());
+        if (!flavor.isPresent()) {
+            throw new PredictorLoadingException(
+                String.format("Attempted to load the %s flavor of the model,"
+                        + " but the model does not have this flavor.",
+                    getFlavorName()));
+        }
+        Optional<String> rootPath = model.getRootPath();
+        if (!rootPath.isPresent()) {
+            throw new PredictorLoadingException(
+                "An internal error occurred while loading the model:"
+                + " the model's root path could not be found. Please ensure that this"
+                + " model was created using `Model.fromRootPath()` or `Model.fromConfigPath()`");
+        }
+        return createPredictor(rootPath.get(), flavor.get());
+    }
+
+    /**
+     * Loads an MLFlow model as a generic predictor that can be used for
+     * inference
+     *
+     * @param modelRootPath The path to the root directory of the MLFlow model
+     *
+     * @throws {@link com.databricks.mlflow.sagemaker.PredictorLoadingException} if
+     * failures are encountered while attempting to load the model from the specified configuration.
+     * This is a generic exception for all loader failure cases
+     */
+    public Predictor load(String modelRootPath) throws PredictorLoadingException {
+        try {
+            return load(Model.fromRootPath(modelRootPath));
```

Comment:
If `Model.fromRootPath` may throw an IOException, we should only wrap that part in the try-catch. Otherwise, only in this code path are we catching IOExceptions and wrapping them -- right now, that seems safe, but it's harder to assert for future versions of this code (e.g., if we add `throws IOException` to the other load).